connect/runtime/src/test/java/org/apache/kafka/connect/integration/MonitorableSourceConnector.java

```diff
@@ -172,5 +235,35 @@ public void stop() {
             stopped = true;
             taskHandle.recordTaskStop();
         }
+
+        private void calculateNextBoundary() {
+            while (nextTransactionBoundary <= seqno) {
+                nextTransactionBoundary += priorTransactionBoundary;
+                priorTransactionBoundary = nextTransactionBoundary - priorTransactionBoundary;
+            }
```

Comment:
Should we mention something about Fibonacci numbers in comment here? I think it's not obvious what we're trying to achieve here.

Reply:
Fair enough, done 👍